dev-tools/mage/manifest/manifest_test.go

```diff
@@ -143,22 +146,77 @@ func TestResolveManifestPackage(t *testing.T) {
 				return
 			}
 
-			urlList, err := resolveManifestPackage(projects[tc.projectName], spec, manifestJson.Version, tc.platform)
+			resolvedPackage, err := ResolveManifestPackage(projects[tc.projectName], spec, manifestJson.Version, tc.platform)
 			require.NoError(t, err)
+			require.NotNil(t, resolvedPackage)
 
-			assert.Len(t, urlList, 3)
-			for _, url := range urlList {
+			assert.Len(t, resolvedPackage.URLs, 3)
+			for _, url := range resolvedPackage.URLs {
 				assert.Contains(t, tc.expectedUrlList, url)
 			}
 		})
 	}
 }
 
-func findBinarySpec(name string) (BinarySpec, bool) {
-	for _, spec := range ExpectedBinaries {
+func findBinarySpec(name string) (packaging.BinarySpec, bool) {
+	for _, spec := range packaging.ExpectedBinaries {
 		if spec.BinaryName == name {
 			return spec, true
 		}
 	}
-	return BinarySpec{}, false
+	return packaging.BinarySpec{}, false
+}
+
+func TestRelaxVersion(t *testing.T) {
+	type args struct {
+		version string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr assert.ErrorAssertionFunc
+	}{
+		{
+			name: "major-minor-patch",
+			args: args{
+				version: "1.2.3",
+			},
+			want:    `1\.2\.(?:0|[1-9]\d*)`,
+			wantErr: assert.NoError,
+		},
+		{
+			name: "major-minor-patch-snapshot",
+			args: args{
+				version: "1.2.3-SNAPSHOT",
+			},
+			want:    `1\.2\.(?:0|[1-9]\d*).*`,
+			wantErr: assert.NoError,
+		},
+		{
+			name: "major-minor-patch-snapshot-buildmeta",
+			args: args{
+				version: "1.2.3-SNAPSHOT+build20250328112233",
```

Comment:
FWIW, the Independent Agent Release never builds any SNAPSHOT builds.  It only builds BC/Staging builds, so no need to check for anything with both `SNAPSHOT` and the `+build` metadata string in it.

Reply:
This is a general test for the `relaxVersion` function which will replace the patch version with a regular expression matching any other patch expression.
Even if Independent Agent Release does not use a version with prerelease and buildmetadata right now, it's worth checking that the resulting regular expression still makes sense.